### PR TITLE
add option to get public key from bech32

### PR DESCRIPTION
**Pull request by dkijania**
For debugging purposes we need to convert public key from bytes. This pr try to solve this.

### CODE REVIEW

File: jcli/src/jcli_lib/key.rs [hunk 1]
```diff
@@ -270,6 +273,21 @@ impl FromBytes {
...
+        if self.public_key {
+            let key_bech32 = match self.key_type {
+                GenPrivKeyType::Ed25519 => bytes_to_pub_key::<Ed25519>(&bytes)?,
+                GenPrivKeyType::Ed25519Bip32 => bytes_to_pub_key::<Ed25519Bip32>(&bytes)?,
+                GenPrivKeyType::Ed25519Extended => unimplemented!(),
+                GenPrivKeyType::SumEd25519_12 => bytes_to_pub_key::<SumEd25519_12>(&bytes)?,
+                GenPrivKeyType::RistrettoGroup2HashDh => {
+                    bytes_to_pub_key::<RistrettoGroup2HashDh>(&bytes)?
+                }
+            };
```
Comment (danielSanchezQ) [inline jcli/src/jcli_lib/key.rs:285]: I would make this a function itself.

Comment (danielSanchezQ) [reply jcli/src/jcli_lib/key.rs:285]: And something like?:

```rust
let bench32_key = if self.public_key {
    process_public_key(self.key_type, bytes)
} else {
    process_private_key(self.key_type, bytes)
};
let mut output = self.output_file.open()?;
writeln!(output, "{}", key_bech32)?;
return Ok(());
```

File: jcli/src/jcli_lib/key.rs [hunk 2]
```diff
@@ -270,6 +273,21 @@ impl FromBytes {
...
+            let mut output = self.output_file.open()?;
+            writeln!(output, "{}", key_bech32)?;
+            return Ok(());
+        }
```
Comment (mzabaluev) [inline jcli/src/jcli_lib/key.rs:289]: Agree with @danielSanchezQ. At least, for better readability, there should be an else branch for the private key output code below.

**Review by mzabaluev — APPROVED** [review]

File: jcli/src/jcli_lib/key.rs [hunk 3]
```diff
@@ -91,6 +91,9 @@ pub struct FromBytes {
 
     #[structopt(flatten)]
     output_file: OutputFile,
+
+    #[structopt(long = "public")]
+    public_key: bool,
```
[the remark this answers is not in the document]
Comment (dkijania) [reply jcli/src/jcli_lib/key.rs:96]: yes

**Review by danielSanchezQ — APPROVED** [review]

**Pull request merged**